app/models/okta_redis/grants.rb

```diff
@@ -8,34 +8,51 @@ module OktaRedis
   class Grants < Model
     CLASS_NAME = 'GrantsService'
 
+    after_initialize do |grants|
+      grants.instance_variable_set(:@activity_log, [])
+    end
+
     def all
-      okta_response.body
+      all_grants = okta_response.body
+      log_message_to_sentry('Getting all grants', :info,
+                            grants_service_activity: @activity_log)
+      all_grants
     end
 
     def delete_grants(grant_ids)
       success = grant_ids.reduce(true) { |memo, grant| delete_grant(grant) && memo }
       OktaRedis::Grants.delete(cache_key) if success
+      log_message_to_sentry('Deleting all grants', :info,
+                            grants_service_activity: @activity_log)
       success
     end
 
     def delete_grant(grant_id)
       delete_response = service.delete_grant(@user.okta_profile.id, grant_id)
+      @activity_log << "Deleting grant id #{grant_id}.  Success? #{delete_response.success?}"
       if delete_response.success?
         true
       else
         log_message_to_sentry("Error deleting grant #{grant_id}", :error,
-                              body: delete_response.body)
+                              {
+                                body: delete_response.body,
+                                grants_service_activity: @activity_log
+                              })
         raise 'Unable to delete grant'
       end
     end
 
     private
 
     def okta_response
-      do_cached_with(key: cache_key) do
+      @activity_log << 'Get okta response including grants'
+      response = do_cached_with(key: cache_key) do
+        @activity_log << 'Fetch new response'
```

Comment:
this is the issue we're ultimately trying to get insight on.  After deleting a grant, we should need to fetch a new response.  BUT if it's something with Okta's service, it wouldn't have changed that much